src/network/networkd-dhcp4.c

```diff
@@ -1449,6 +1461,96 @@ int config_parse_dhcp_request_options(
         return 0;
 }
 
+int config_parse_dhcp_send_options(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = data;
+        const char *x;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                network->dhcp_send_options = ordered_hashmap_free_with_destructor(network->dhcp_send_options, sd_dhcp_option_unref);
+                return 0;
+        }
+
+        r = ordered_hashmap_ensure_allocated(&network->dhcp_send_options, &trivial_hash_ops);
+        if (r < 0)
+                return log_oom();
+
+        for (x = rvalue;;) {
+                _cleanup_free_ char *word = NULL;
+                sd_dhcp_option *p;
+                uint32_t u;
+                char *split;
+
+                r = extract_first_word(&x, &word, NULL, EXTRACT_UNQUOTE);
+                if (r == 0)
+                        return 0;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to extract first word, ignoring: %s", rvalue);
+                        return 0;
+                }
+
+                split = strchr(word, ':');
+                if (!split)
```

Comment:
If there is no ":" then the word is completely ignored. This is wrong.

Reply:
Why send empty values ? This should be ignored.

Reply:
The syntax is "a:b". If ":" is missing, then the entry does not have valid syntax.

Reply:
yes updated. thanks